reflect.go

```diff
@@ -274,21 +275,25 @@ func (r *Reflector) reflectTypeToSchema(definitions Definitions, t reflect.Type)
 			return rt
 		}
 
+		prop, err := json.Marshal(r.reflectTypeToSchema(definitions, t.Elem()))
+		if err != nil {
+			panic(err)
+		}
+
+		if bytes.Equal(prop, []byte("{}")) {
+			prop = []byte("true")
+		}
+
 		rt := &Type{
-			Type: "object",
-			PatternProperties: map[string]*Type{
-				".*": r.reflectTypeToSchema(definitions, t.Elem()),
-			},
+			Type:                 "object",
```

Comment:
This is [not semantically equivalent](http://json-schema.org/understanding-json-schema/reference/object.html#pattern-properties) - your change allows any property of any type, while the previous approach applied a sub-schema to the values of the map.